rust/connlib/tunnel/src/client.rs

```diff
@@ -1183,6 +1181,66 @@ mod tests {
         )
     }
 
+    #[test]
+    fn add_resources_works() {
```

Comment:
I think what you are testing here would be perfect for a property-based test. Take some inspiration from

https://github.com/firezone/firezone/blob/ab598eff912375b694b15aafaec545ef6286c34e/rust/relay/src/proptest.rs#L12-L59

and

https://github.com/firezone/firezone/blob/ab598eff912375b694b15aafaec545ef6286c34e/rust/relay/src/server/channel_data.rs#L84-L107

Reply:
You'll have to slightly change the way you think about your test. Instead of asserting on fixed strings, you want to assert on "properties" such as "every CIDR resources results in a route".

Reply:
#4355